design-proposals/vm-rollout-strategy/vm-rollout-strategy.md

```diff
@@ -0,0 +1,195 @@
+# Overview
+
+Today changes to a VM object (i.e. changing VM.spec.template) get applied, but only take effect upon the next restart - we call these pending changes “staged”. This is a proposal to introduce a rollout strategy field to control how changes to the VM object will propagate to the VirtualMachineInstance object.
```

Comment:
This was one of the proposed features for VMPools, but we didn't get to implementing it yet.   Can you add to your design how VMPools could leverage this feature on the VM API?

Reply:
That's actually similar to the `DeploymentStrategy` of Deployments - which can be RollingUpdate or Recreate.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#deploymentstrategy-v1-apps

One question that we have two streams of opinions about is:
Do we expose the different VM update strategies (liveupdate or (current) restart) on the VM or only on the Cluster level?

For VMPools I'd assume that we need a VM level control.
The VMPools either change the behavior if liveupdate is enabeld or not, or if liveupdate is enabled, then on VMPools the strategy can be selected.

Reply:
Let#s defer the vmpool discussion pls